types/foundry/client/applications/apps/document-sheet-config.d.mts

```diff
@@ -0,0 +1,154 @@
+    static override DEFAULT_OPTIONS: DeepPartial<DocumentSheetConfiguration>;
+
+    static override PARTS: Record<string, HandlebarsTemplatePart>;
+
+    /* -------------------------------------------- */
+    /*  Rendering                                   */
+    /* -------------------------------------------- */
+
+    override _preparePartContext(partId: string, context: object, options: HandlebarsRenderOptions): Promise<object>;
+
+    /** Prepare render context for the footer part. */
+    protected _prepareFooterContext(context: ApplicationRenderContext, options: HandlebarsRenderOptions): Promise<void>;
+
+    /** Prepare render context for the form part. */
+    protected _prepareFormContext(context: ApplicationRenderContext, options: HandlebarsRenderOptions): Promise<void>;
+
+    /* -------------------------------------------- */
+    /*  Event Listeners & Handlers                  */
+    /* -------------------------------------------- */
+
+    override _onChangeForm(formConfig: ApplicationFormConfiguration, event: Event): void;
+
+    /* -------------------------------------------- */
+    /*  Helpers                                     */
+    /* -------------------------------------------- */
+
+    /**
+     * Marshal information on the available sheet classes for a given document type and sub-type, and format it for
+     * display.
+     * @param documentName  The Document type.
+     * @param [subType]     The Document sub-type, if applicable.
+     */
+    static getSheetClassesForSubType(
+        documentName: string,
+        subType?: string,
+    ): { sheetClasses: Record<string, string>; defaultClasses: Record<string, string>; defaultClass: string };
+
+    /**
+     * Retrieve the user's theme preference for the given Document.
+     * @param document  The Document.
+     * @returns         The theme identifier, or a blank string if the user has no preference.
+     */
+    static getSheetThemeForDocument(document: ClientDocument): string;
+
+    /* -------------------------------------------- */
+    /*  Sheet Configuration API                     */
+    /* -------------------------------------------- */
+
+    /** Initialize the configured sheet preferences for Documents which support dynamic sheet assignment. */
+    static initializeSheets(): Promise<void>;
+
+    /**
+     * Register a sheet class as a candidate to be used to display Documents of a given type.
+     * @param documentClass  The Document class to register a new sheet for.
+     * @param scope          A unique namespace scope for this sheet.
+     * @param sheetClass     An Application class used to render the sheet.
+     * @param options        Sheet registration configuration options.
+     */
+    static registerSheet(
+        documentClass: typeof ClientDocument,
+        scope: string,
+        sheetClass: typeof Application | typeof ApplicationV2,
```

Comment:
Probably need to make these `AbstractConstructorOf<Application>`, etc. since they're generic